src/test/java/org/springframework/cloud/dataflow/acceptance/test/AbstractStreamTests.java

```diff
@@ -355,56 +364,61 @@ protected boolean waitForLogEntry(Application app, String... entries) {
 		logger.info("Looking for '" + StringUtils.arrayToCommaDelimitedString(entries) + "' in logfile for "
 				+ app.getDefinition());
 		final long timeout = System.currentTimeMillis() + (configurationProperties.getMaxWaitTime() * 1000);
-		boolean exists = false;
-		String instance = "?";
-		Map<String, String> logData = new HashMap<>();
+		List<Log> logs = null;
+		LogRetriever logRetriever = isCloudFoundry() ? new CloudFoundryLogRetriever(app) : new DefaultLogRetriever(app);
 
-		while (!exists && System.currentTimeMillis() < timeout) {
+		while (System.currentTimeMillis() < timeout) {
 			try {
 				Thread.sleep(configurationProperties.getDeployPauseTime() * 1000);
 			}
 			catch (InterruptedException e) {
 				Thread.currentThread().interrupt();
 				throw new IllegalStateException(e.getMessage(), e);
 			}
-			for (String appInstance : app.getInstanceUrls().keySet()) {
-				if (!exists) {
-					logger.info("Requesting log for app " + appInstance);
-					String log = getLog(app.getInstanceUrls().get(appInstance));
-					logData.put(appInstance, log);
-					if (log != null) {
-						if (Stream.of(entries).allMatch(log::contains)) {
-							exists = true;
-							instance = appInstance;
-						}
-					}
-					else {
-						logger.info("Polling to get log file. Remaining poll time = "
-								+ Long.toString((timeout - System.currentTimeMillis()) / 1000) + " seconds.");
+			try {
```

Comment:
The InterruptedException and the IOException?

Probably.  The 2 reasons the CloudFoundryLogRetriever::retrieveLogs would I/O error:
* command not found
* error reading command output
The first should be changed to error in a way the test exists.
The second _could_ resolved itself on subsequent tries.

However this could be moot pending the CF Java API conversation

Reply:
Changed so the the first case kills the case.

In the second case, rather than throw exception, an error is logged and the command stdout (or stderr) string is deemed `null`.